### PR TITLE
common: nicer stat usage

### PR DESCRIPTION
**Pull request by tobsan**
Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: common/softwarecontainer-common.cpp [hunk 1]
```diff
@@ -35,20 +35,22 @@
 namespace softwarecontainer {
     LOG_DECLARE_DEFAULT_CONTEXT(defaultLogContext, "MAIN", "Main context");
 
-struct stat getStat(const std::string &path)
+bool getStat(const std::string &path, struct stat &st)
 {
-    struct stat st;
-    memset(&st, 0, sizeof(st));
-
     if (stat(path.c_str(), &st) == 0) {
```
Comment (jonte) [inline common/softwarecontainer-common.cpp]: Maybe shorten the function to:

`return stat(path.c_str(), &st) == 0;`

**Comment by breakreturn** [comment]
No tests changed? I assume we don't have any then. Might be a good idea to add unit tests for this?

File: common/softwarecontainer-common.cpp [hunk 2]
```diff
@@ -75,22 +77,38 @@ bool isDirectoryEmpty(const std::string &path) {
 
 bool isFile(const std::string &path)
 {
-    return S_ISREG(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISREG(st.st_mode);
```
Comment (jonte) [inline common/softwarecontainer-common.cpp:79]: I might be nitpicking, but maybe `return getStat(path, st) && S_ISREG(st.st_mode);`

I'm OK w/ the semantics here though.

File: common/softwarecontainer-common.cpp [hunk 3]
```diff
@@ -75,22 +77,38 @@ bool isDirectoryEmpty(const std::string &path) {
...
-    return S_ISSOCK(getStat(path).st_mode);
+    struct stat st;
+    if (getStat(path, st)) {
+        return S_ISSOCK(st.st_mode);
+    }
+
+    return false;
 }
 
 bool existsInFileSystem(const std::string &path)
```
Comment (jonte) [inline common/softwarecontainer-common.cpp:105]: This seems like a weird call overall.. Why would you ever check this? What you want is to stat if the file exists and is in some way accessible, right?

**Comment by tobsan** [comment]
@breakreturn we already have unit tests for this. Since this only changes the internals of the functions, those tests are still valid and running.

**Review by jonte — APPROVED** [review]
Looks good, I have some nitpicking if you're interested in polishing.

File: common/softwarecontainer-common.cpp (hunk 3 above)
Comment (tobsan) [reply common/softwarecontainer-common.cpp:105]: Yes.

**Pull request merged**